-1

**Failed Tests**:   Build  ClangBuild 
**Size**: This PR adds an extra 20KB to repository
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c18ff3/53849/summary.html
**COMMIT**: fbe9efac5232cf9cd88bd1aceee4c869cc2b09cb
**CMSSW**: CMSSW_20_0_X_2026-06-10-2300/el8_amd64_gcc13
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/51194/53849/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Failed Build

I found compilation error when building: 

<pre>>> Compiling  src/Geometry/CaloTopology/src/HcalTopology.cc
/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02945/el8_amd64_gcc13/external/gcc/13.4.0-6908cfdf803923e783448096ca4f0923/bin/c++ -c -DCMS_MICRO_ARCH='x86-64-v3' -DGNU_GCC -D_GNU_SOURCE -DTBB_USE_GLIBCXX_VERSION=130400 -DTBB_SUPPRESS_DEPRECATED_MESSAGES -DTBB_PREVIEW_RESUMABLE_TASKS=1 -DTBB_PREVIEW_TASK_GROUP_EXTENSIONS=1 -DBOOST_SPIRIT_THREADSAFE -DPHOENIX_THREADSAFE -DBOOST_MATH_DISABLE_STD_FPCLASSIFY -DBOOST_UUID_RANDOM_PROVIDER_FORCE_POSIX -DBOOST_MPL_IGNORE_PARENTHESES_WARNING -DGNU_GCC -DG4V9 -DGNU_GCC -DG4V9 -DDD4HEP_USE_GEANT4_UNITS=1 -DCMSSW_GIT_HASH='CMSSW_20_0_X_2026-06-10-2300' -DPROJECT_NAME='CMSSW' -DPROJECT_VERSION='CMSSW_20_0_X_2026-06-10-2300' -Isrc -Ipoison -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02945/el8_amd64_gcc13/cms/cmssw-patch/CMSSW_20_0_X_2026-06-10-2300/src -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02945/el8_amd64_gcc13/external/dd4hep/v01-31-0x-1f66834f6c77bf4f3f54b5b8438c37be/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02945/el8_amd64_gcc13/external/geant4/11.4.1-9f69da9c322a053dd7281502844a2601/include/Geant4 -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02945/el8_amd64_gcc13/external/geant4/11.4.1-9f69da9c322a053dd7281502844a2601/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02945/el8_amd64_gcc13/external/pcre/8.43-7f0edb13c32a66555b25b5433dd2ceed/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02945/el8_amd64_gcc13/external/boost/1.80.0-439e0a2183f4d7ca3604168e474224c5/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02945/el8_amd64_gcc13/external/bz2lib/1.0.8-353e2ef1c2f1d06b28a8938defa197bc/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02945/el8_amd64_gcc13/external/clhep/2.4.7.2-e2fcd4be043de62aca9bcd008063cdf8/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02945/el8_amd64_gcc13/external/expat/2.7.1-fd861297a37b8f9a1d9e1c6e2937faaa/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02945/el8_amd64_gcc13/external/gsl/2.6-9011a41928244b609ca4c22c439b3fef/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02945/el8_amd64_gcc13/external/libuuid/2.40-d94705b84648fd294d63bf9de23bc03b/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02945/el8_amd64_gcc13/lcg/root/6.36.13-7fe36d08dd9b9758c207859463cc3525/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02945/el8_amd64_gcc13/external/tbb/v2022.3.0-daad917f8c7fd382a88c95b720f7e5f7/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02945/el8_amd64_gcc13/external/vecgeom/v2.0.0-0af9277edb52b71b35420b5f67c4ed2e/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02945/el8_amd64_gcc13/external/vecgeom/v2.0.0-0af9277edb52b71b35420b5f67c4ed2e/include/VecGeom -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02945/el8_amd64_gcc13/external/xerces-c/3.1.3-ab719034dc2061fa61b09353203a0c2d/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02945/el8_amd64_gcc13/external/xz/5.6.4-ea99661ac7ca423b4d74fd650284b41f/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02945/el8_amd64_gcc13/external/zlib/1.3.2-53c061acc706fe8f28fd9477d9981c20/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02945/el8_amd64_gcc13/external/eigen/c1d637433e3b3f9012b226c2c9125c494b470ae6-98caed40e4d2e9af234831b4eafb854a/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02945/el8_amd64_gcc13/external/eigen/c1d637433e3b3f9012b226c2c9125c494b470ae6-98caed40e4d2e9af234831b4eafb854a/include/eigen3 -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02945/el8_amd64_gcc13/external/md5/1.0.0-26057075013e190e56dad37d35219376/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02945/el8_amd64_gcc13/external/OpenBLAS/0.3.27-da4a3c2bb8ae43f3913a4a44acdb1b50/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02945/el8_amd64_gcc13/external/tinyxml2/6.2.0-ccad39708675ec9caf4eb88d12976cd5/include -O3 -pthread -pipe -Werror=main -Werror=pointer-arith -Werror=overlength-strings -Wno-vla -Werror=overflow -std=c++20 -ftree-vectorize -Werror=array-bounds -Werror=format-contains-nul -Werror=type-limits -fvisibility-inlines-hidden -fno-math-errno --param vect-max-version-for-alias-checks=50 -Xassembler --compress-debug-sections -Wno-error=array-bounds -Warray-bounds -fuse-ld=bfd -march=x86-64-v3 -felide-constructors -fmessage-length=0 -Wall -Wno-non-template-friend -Wno-long-long -Wreturn-type -Wextra -Wpessimizing-move -Wclass-memaccess -Wno-cast-function-type -Wno-unused-but-set-parameter -Wno-ignored-qualifiers -Wno-unused-parameter -Wunused -Wparentheses -Werror=return-type -Werror=unused-value -Werror=unused-result -Werror=unused-label -Werror=address -Werror=format -Werror=sign-compare -Werror=write-strings -Werror=delete-non-virtual-dtor -Werror=strict-aliasing -Werror=narrowing -Werror=unused-but-set-variable -Werror=reorder -Werror=unused-variable -Werror=conversion-null -Werror=return-local-addr -Wnon-virtual-dtor -Werror=switch -fdiagnostics-show-option -Wno-unused-local-typedefs -Wno-attributes -Wno-psabi -DEIGEN_DONT_PARALLELIZE -DEIGEN_MAX_ALIGN_BYTES=64 -Wno-error=unused-variable -ftls-model=global-dynamic -pthread -DBOOST_DISABLE_ASSERTS -flto=auto -fipa-icf -flto-odr-type-merging -fno-fat-lto-objects -Wodr -fPIC -MMD -MF tmp/el8_amd64_gcc13/src/Geometry/CaloTopology/src/GeometryCaloTopology/HcalTopology.cc.d src/Geometry/CaloTopology/src/HcalTopology.cc -o tmp/el8_amd64_gcc13/src/Geometry/CaloTopology/src/GeometryCaloTopology/HcalTopology.cc.o
>> Compiling  src/Geometry/CaloTopology/src/HcalTopologyRestrictionParser.cc
/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02945/el8_amd64_gcc13/external/gcc/13.4.0-6908cfdf803923e783448096ca4f0923/bin/c++ -c -DCMS_MICRO_ARCH='x86-64-v3' -DGNU_GCC -D_GNU_SOURCE -DTBB_USE_GLIBCXX_VERSION=130400 -DTBB_SUPPRESS_DEPRECATED_MESSAGES -DTBB_PREVIEW_RESUMABLE_TASKS=1 -DTBB_PREVIEW_TASK_GROUP_EXTENSIONS=1 -DBOOST_SPIRIT_THREADSAFE -DPHOENIX_THREADSAFE -DBOOST_MATH_DISABLE_STD_FPCLASSIFY -DBOOST_UUID_RANDOM_PROVIDER_FORCE_POSIX -DBOOST_MPL_IGNORE_PARENTHESES_WARNING -DGNU_GCC -DG4V9 -DGNU_GCC -DG4V9 -DDD4HEP_USE_GEANT4_UNITS=1 -DCMSSW_GIT_HASH='CMSSW_20_0_X_2026-06-10-2300' -DPROJECT_NAME='CMSSW' -DPROJECT_VERSION='CMSSW_20_0_X_2026-06-10-2300' -Isrc -Ipoison -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02945/el8_amd64_gcc13/cms/cmssw-patch/CMSSW_20_0_X_2026-06-10-2300/src -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02945/el8_amd64_gcc13/external/dd4hep/v01-31-0x-1f66834f6c77bf4f3f54b5b8438c37be/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02945/el8_amd64_gcc13/external/geant4/11.4.1-9f69da9c322a053dd7281502844a2601/include/Geant4 -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02945/el8_amd64_gcc13/external/geant4/11.4.1-9f69da9c322a053dd7281502844a2601/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02945/el8_amd64_gcc13/external/pcre/8.43-7f0edb13c32a66555b25b5433dd2ceed/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02945/el8_amd64_gcc13/external/boost/1.80.0-439e0a2183f4d7ca3604168e474224c5/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02945/el8_amd64_gcc13/external/bz2lib/1.0.8-353e2ef1c2f1d06b28a8938defa197bc/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02945/el8_amd64_gcc13/external/clhep/2.4.7.2-e2fcd4be043de62aca9bcd008063cdf8/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02945/el8_amd64_gcc13/external/expat/2.7.1-fd861297a37b8f9a1d9e1c6e2937faaa/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02945/el8_amd64_gcc13/external/gsl/2.6-9011a41928244b609ca4c22c439b3fef/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02945/el8_amd64_gcc13/external/libuuid/2.40-d94705b84648fd294d63bf9de23bc03b/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02945/el8_amd64_gcc13/lcg/root/6.36.13-7fe36d08dd9b9758c207859463cc3525/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02945/el8_amd64_gcc13/external/tbb/v2022.3.0-daad917f8c7fd382a88c95b720f7e5f7/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02945/el8_amd64_gcc13/external/vecgeom/v2.0.0-0af9277edb52b71b35420b5f67c4ed2e/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02945/el8_amd64_gcc13/external/vecgeom/v2.0.0-0af9277edb52b71b35420b5f67c4ed2e/include/VecGeom -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02945/el8_amd64_gcc13/external/xerces-c/3.1.3-ab719034dc2061fa61b09353203a0c2d/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02945/el8_amd64_gcc13/external/xz/5.6.4-ea99661ac7ca423b4d74fd650284b41f/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02945/el8_amd64_gcc13/external/zlib/1.3.2-53c061acc706fe8f28fd9477d9981c20/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02945/el8_amd64_gcc13/external/eigen/c1d637433e3b3f9012b226c2c9125c494b470ae6-98caed40e4d2e9af234831b4eafb854a/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02945/el8_amd64_gcc13/external/eigen/c1d637433e3b3f9012b226c2c9125c494b470ae6-98caed40e4d2e9af234831b4eafb854a/include/eigen3 -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02945/el8_amd64_gcc13/external/md5/1.0.0-26057075013e190e56dad37d35219376/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02945/el8_amd64_gcc13/external/OpenBLAS/0.3.27-da4a3c2bb8ae43f3913a4a44acdb1b50/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02945/el8_amd64_gcc13/external/tinyxml2/6.2.0-ccad39708675ec9caf4eb88d12976cd5/include -O3 -pthread -pipe -Werror=main -Werror=pointer-arith -Werror=overlength-strings -Wno-vla -Werror=overflow -std=c++20 -ftree-vectorize -Werror=array-bounds -Werror=format-contains-nul -Werror=type-limits -fvisibility-inlines-hidden -fno-math-errno --param vect-max-version-for-alias-checks=50 -Xassembler --compress-debug-sections -Wno-error=array-bounds -Warray-bounds -fuse-ld=bfd -march=x86-64-v3 -felide-constructors -fmessage-length=0 -Wall -Wno-non-template-friend -Wno-long-long -Wreturn-type -Wextra -Wpessimizing-move -Wclass-memaccess -Wno-cast-function-type -Wno-unused-but-set-parameter -Wno-ignored-qualifiers -Wno-unused-parameter -Wunused -Wparentheses -Werror=return-type -Werror=unused-value -Werror=unused-result -Werror=unused-label -Werror=address -Werror=format -Werror=sign-compare -Werror=write-strings -Werror=delete-non-virtual-dtor -Werror=strict-aliasing -Werror=narrowing -Werror=unused-but-set-variable -Werror=reorder -Werror=unused-variable -Werror=conversion-null -Werror=return-local-addr -Wnon-virtual-dtor -Werror=switch -fdiagnostics-show-option -Wno-unused-local-typedefs -Wno-attributes -Wno-psabi -DEIGEN_DONT_PARALLELIZE -DEIGEN_MAX_ALIGN_BYTES=64 -Wno-error=unused-variable -ftls-model=global-dynamic -pthread -DBOOST_DISABLE_ASSERTS -flto=auto -fipa-icf -flto-odr-type-merging -fno-fat-lto-objects -Wodr -fPIC -MMD -MF tmp/el8_amd64_gcc13/src/Geometry/CaloTopology/src/GeometryCaloTopology/HcalTopologyRestrictionParser.cc.d src/Geometry/CaloTopology/src/HcalTopologyRestrictionParser.cc -o tmp/el8_amd64_gcc13/src/Geometry/CaloTopology/src/GeometryCaloTopology/HcalTopologyRestrictionParser.cc.o
src/Geometry/CaloTopology/src/HGCalTopology.cc: In constructor 'HGCalTopology::HGCalTopology(const HGCalDDDConstants&, int)':
src/Geometry/CaloTopology/src/HGCalTopology.cc:46:61: error: 'const class HGCalDDDConstants' has no member named 'endcaps'
   46 |   kSizeForDenseIndexing = static_cast<unsigned int>(hdcons_.endcaps() * kHGhalf_);
      |                                                             ^~~~~~~
gmake: *** [tmp/el8_amd64_gcc13/src/Geometry/CaloTopology/src/GeometryCaloTopology/HGCalTopology.cc.o] Error 1
>> Building shared library tmp/el8_amd64_gcc13/src/Geometry/CaloTopology/src/GeometryCaloTopology/libGeometryCaloTopology.so
/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02945/el8_amd64_gcc13/external/gcc/13.4.0-6908cfdf803923e783448096ca4f0923/bin/c++ -O3 -pthread -pipe -Werror=main -Werror=pointer-arith -Werror=overlength-strings -Wno-vla -Werror=overflow -std=c++20 -ftree-vectorize -Werror=array-bounds -Werror=format-contains-nul -Werror=type-limits -fvisibility-inlines-hidden -fno-math-errno --param vect-max-version-for-alias-checks=50 -Xassembler --compress-debug-sections -Wno-error=array-bounds -Warray-bounds -fuse-ld=bfd -march=x86-64-v3 -felide-constructors -fmessage-length=0 -Wall -Wno-non-template-friend -Wno-long-long -Wreturn-type -Wextra -Wpessimizing-move -Wclass-memaccess -Wno-cast-function-type -Wno-unused-but-set-parameter -Wno-ignored-qualifiers -Wno-unused-parameter -Wunused -Wparentheses -Werror=return-type -Werror=unused-value -Werror=unused-result -Werror=unused-label -Werror=address -Werror=format -Werror=sign-compare -Werror=write-strings -Werror=delete-non-virtual-dtor -Werror=strict-aliasing -Werror=narrowing -Werror=unused-but-set-variable -Werror=reorder -Werror=unused-variable -Werror=conversion-null -Werror=return-local-addr -Wnon-virtual-dtor -Werror=switch -fdiagnostics-show-option -Wno-unused-local-typedefs -Wno-attributes -Wno-psabi -DEIGEN_DONT_PARALLELIZE -DEIGEN_MAX_ALIGN_BYTES=64 -Wno-error=unused-variable -ftls-model=global-dynamic -pthread -DBOOST_DISABLE_ASSERTS -flto=auto -fipa-icf -flto-odr-type-merging -fno-fat-lto-objects -Wodr -shared -Wl,-E    -Wl,-z,defs     tmp/el8_amd64_gcc13/src/Geometry/CaloTopology/src/GeometryCaloTopology/CaloDirection.cc.o tmp/el8_amd64_gcc13/src/Geometry/CaloTopology/src/GeometryCaloTopology/CaloSubdetectorTopology.cc.o tmp/el8_amd64_gcc13/src/Geometry/CaloTopology/src/GeometryCaloTopology/CaloTopology.cc.o tmp/el8_amd64_gcc13/src/Geometry/CaloTopology/src/GeometryCaloTopology/CaloTowerConstituentsMap.cc.o tmp/el8_amd64_gcc13/src/Geometry/CaloTopology/src/GeometryCaloTopology/CaloTowerTopology.cc.o tmp/el8_amd64_gcc13/src/Geometry/CaloTopology/src/GeometryCaloTopology/ES_CaloSubdetectorTopology.cc.o tmp/el8_amd64_gcc13/src/Geometry/CaloTopology/src/GeometryCaloTopology/ES_CaloTopogy.cc.o tmp/el8_amd64_gcc13/src/Geometry/CaloTopology/src/GeometryCaloTopology/ES_Topologies.cc.o tmp/el8_amd64_gcc13/src/Geometry/CaloTopology/src/GeometryCaloTopology/EcalBarrelHardcodedTopology.cc.o tmp/el8_amd64_gcc13/src/Geometry/CaloTopology/src/GeometryCaloTopology/EcalBarrelTopology.cc.o tmp/el8_amd64_gcc13/src/Geometry/CaloTopology/src/GeometryCaloTopology/EcalEndcapHardcodedTopology.cc.o tmp/el8_amd64_gcc13/src/Geometry/CaloTopology/src/GeometryCaloTopology/EcalEndcapTopology.cc.o tmp/el8_amd64_gcc13/src/Geometry/CaloTopology/src/GeometryCaloTopology/EcalPreshowerTopology.cc.o tmp/el8_amd64_gcc13/src/Geometry/CaloTopology/src/GeometryCaloTopology/EcalTrigTowerConstituentsMap.cc.o tmp/el8_amd64_gcc13/src/Geometry/CaloTopology/src/GeometryCaloTopology/HGCalNeighbourFinder.cc.o tmp/el8_amd64_gcc13/src/Geometry/CaloTopology/src/GeometryCaloTopology/HGCalTBTopology.cc.o tmp/el8_amd64_gcc13/src/Geometry/CaloTopology/src/GeometryCaloTopology/HGCalTopology.cc.o tmp/el8_amd64_gcc13/src/Geometry/CaloTopology/src/GeometryCaloTopology/HcalTopology.cc.o tmp/el8_amd64_gcc13/src/Geometry/CaloTopology/src/GeometryCaloTopology/HcalTopologyRestrictionParser.cc.o -o tmp/el8_amd64_gcc13/src/Geometry/CaloTopology/src/GeometryCaloTopology/libGeometryCaloTopology.so -Wl,-E -Wl,--hash-style=gnu -Wl,--as-needed -Wl,-z,noexecstack -L/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_20_0_X_2026-06-10-2300/biglib/el8_amd64_gcc13 -L/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_20_0_X_2026-06-10-2300/lib/el8_amd64_gcc13 -L/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02945/el8_amd64_gcc13/cms/cmssw-patch/CMSSW_20_0_X_2026-06-10-2300/biglib/el8_amd64_gcc13 -L/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02945/el8_amd64_gcc13/cms/cmssw-patch/CMSSW_20_0_X_2026-06-10-2300/lib/el8_amd64_gcc13 -L/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02945/el8_amd64_gcc13/cms/cmssw-patch/CMSSW_20_0_X_2026-06-10-2300/external/el8_amd64_gcc13/lib -L/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02945/el8_amd64_gcc13/cms/cmssw/CMSSW_20_0_X_2026-06-07-0000/lib/el8_amd64_gcc13 -L/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02945/el8_amd64_gcc13/cms/cmssw/CMSSW_20_0_X_2026-06-07-0000/biglib/el8_amd64_gcc13 -L/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_20_0_X_2026-06-10-2300/static/el8_amd64_gcc13 -L/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02945/el8_amd64_gcc13/cms/cmssw-patch/CMSSW_20_0_X_2026-06-10-2300/static/el8_amd64_gcc13 -L/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02945/el8_amd64_gcc13/cms/cmssw/CMSSW_20_0_X_2026-06-07-0000/static/el8_amd64_gcc13 -lGeometryHGCalTBCommonData -lGeometryHGCalCommonData -lGeometryHcalCommonData -lDetectorDescriptionDDCMS -lFWCoreFramework -lFWCoreVersion -lDetectorDescriptionRegressionTest -lGeometryCaloGeometry -lDataFormatsCaloTowers -lDetectorDescriptionParser -lDataFormatsCandidate -lDetectorDescriptionCore -lCondFormatsGeometryObjects -lDataFormatsEcalDetId -lDataFormatsForwardDetId -lDataFormatsGeometryVector -lDataFormatsHcalDetId -lSimDataFormatsCaloHit -lDataFormatsDetId -lDataFormatsMath -lFWCoreCommon -lFWCoreServiceRegistry -lDataFormatsCommon -lFWCoreParameterSet -lFWCoreAbstractServices -lFWCoreMessageLogger -lDataFormatsProvenance -lFWCoreConcurrency -lFWCorePluginManager -lFWCoreReflection -lCondFormatsSerialization -lFWCoreUtilities -lSimDataFormatsEncodedEventId -lUtilitiesXerces -lDDAlign -lDDCond -lDDCore -lDDParsers -lPhysics -lHist -lMatrix -lGenVector -lMathMore -lTree -lNet -lGeom -lThread -lG4digits_hits -lG4error_propagation -lG4event -lG4geometry -lG4global -lG4graphics_reps -lG4intercoms -lG4interfaces -lG4materials -lG4parmodels -lG4particles -lG4geomtext -lG4mctruth -lG4gdml -lG4physicslists -lG4processes_core -lG4processes_hadronic -lG4readout -lG4run -lG4tracking -lG4track -lG4analysis -lG4ptl -lMathCore -lRIO -lboost_program_options -lboost_serialization -lCore -lboost_thread -lboost_date_time -lCLHEP -lpcre -lvecgeom -lbz2 -lexpat -lgsl -luuid -ltbb -lxerces-c -llzma -lz -lcms-md5 -lopenblas -lcrypt -ldl -lrt -ltinyxml2
</pre>

## Failed Clang Build

I found compilation error while trying to compile with clang. Command used:
```
USER_CUDA_FLAGS='--expt-relaxed-constexpr' USER_CXXFLAGS='-Wno-register -fsyntax-only' /usr/bin/time -v scram build -k -j 32 COMPILER='llvm compile'

```


<pre>>> Entering Package Geometry/CaloTopology
>> Entering Package Geometry/HGCalCommonData
>> Entering Package Geometry/HGCalGeometry
>> Entering Package SimG4CMS/Calo
>> Compile sequence completed for CMSSW CMSSW_20_0_X_2026-06-10-2300
gmake: *** [There are compilation/build errors. Please see the detail log above.] Error 1
Command exited with non-zero status 1
	Command being timed: "scram build -k -j 32 COMPILER=llvm compile BUILD_LOG=yes"
	User time (seconds): 544.66
	System time (seconds): 66.29
	Percent of CPU this job got: 1352%
</pre>

--- a/Geometry/HGCalCommonData/interface/HGCalDDDConstants.h
+++ b/Geometry/HGCalCommonData/interface/HGCalDDDConstants.h
@@ -78,6 +78,7 @@ public:
   }
   double distFromEdgeHex(double x, double y, double z) const;
   double distFromEdgeTrap(double x, double y, double z) const;
+  int endcaps() const { return (coldBoxMode() ? 1 : 2); }
   void etaPhiFromPosition(const double x,
                           const double y,
                           const double z,

--- a/Geometry/HGCalCommonData/src/HGCalParametersFromDD.cc
+++ b/Geometry/HGCalCommonData/src/HGCalParametersFromDD.cc
@@ -97,6 +97,16 @@ bool HGCalParametersFromDD::build(const DDCompactView* cpv,
           (php.mode_ == HGCalGeometryMode::Hexagon8CalibCell) || (php.mode_ == HGCalGeometryMode::Hexagon8FineCell)) {
         php.useSimWt_ = static_cast<int>(getDDDValue("UseSimWt", sv));
         php.layerRotation_ = getDDDValue("LayerRotation", sv);
+        php.coldBoxMode_ = static_cast<int>(getDDDValue("ColdBoxMode", sv));
+        if (php.coldBoxMode_ > 0) {  // Cold Box
+          for (unsigned int k = 0; k < 20; k++) {
+            std::string sLRot = "LayerRotation" + std::to_string(k + 1);
+            php.coldBoxRots_.push_back(getDDDValue(sLRot.c_str(), sv));
+#ifdef EDM_ML_DEBUG
+            edm::LogVerbatim("HGCalGeom") << "HGParameterFromDD rots[" << k << "] " << php.coldBoxRots_[k];
+#endif
+          }
+        }
       }
       if ((php.mode_ == HGCalGeometryMode::Hexagon8CalibCell) || (php.mode_ == HGCalGeometryMode::Hexagon8FineCell)) {
       }

